### PR TITLE
Add parser support for arrays 

### PR DESCRIPTION
**Pull request by lukasberglund**
#38

### CODE REVIEW

File: test/Parser/ParserSpec.hs [hunk 1]
```diff
@@ -40,6 +40,7 @@ spec = do
         , "foo:uint64 = 10 + 20 * 30!"
         , "(declare ($foo : uint64) (+ 10 (* 20 30)))")
       , ("with arbitrary whitespace", "a  : \t void!", "(declare ($a : void))")
+      , ("with an array type", "a : [int]!", "(declare ($a : [int]))")
```
Comment (ayazhafiz) [inline test/Parser/ParserSpec.hs:43]: can you add an assignment test too?

File: src/Grammar.hs [hunk 2]
```diff
@@ -60,6 +62,7 @@ instance Read Type where
 data Expression
   = Number String
   | String String
+  | List [Expression]
```
Comment (ayazhafiz) [inline src/Grammar.hs:65]: I don't think an expression should be able to be a list of expressions.

File: test/Parser/ParserSpec.hs [hunk 3]
```diff
@@ -36,10 +42,14 @@ spec = do
       , ( "with an initializer"
         , "foo:uint64 = 10!"
         , "(declare ($foo : uint64) 10)")
+      , ( "with an initializer and an array type"
+        , "a : [double] = []!"
+        , "(declare ($a : [double]) [])")
```
Comment (ayazhafiz) [inline test/Parser/ParserSpec.hs:47]: nit: put this under `describe "arrays"`

File: src/Grammar.hs (hunk 2 above)
Comment (lukasberglund) [reply src/Grammar.hs:65]: This is to initialize arrays.

**Pull request merged**